rdflib/tools/defined_namespace_creator.py

```diff
@@ -0,0 +1,182 @@
+"""
+This rdflib Python script creates a DefinedNamespace Python file from a given RDF file
+
+It is a very simple script: it finds all things defined in the RDF file within a given
+namespace:
+
+    <thing> a ?x
+
+    where ?x is anything and <thing> starts with the given namespace
+
+Nicholas J. Car, Dec, 2021
+"""
+import sys
+from pathlib import Path
+import argparse
+import datetime
+
+sys.path.append(str(Path(__file__).parent.absolute().parent.parent))
+
+from rdflib import Graph, util
+from rdflib.namespace import DCTERMS, OWL, RDFS, SKOS
+
+
+def get_input_format(file_path):
+    input_format = util.guess_format(str(file_path))
+    if input_format is None:
```

Comment:
I think the latest util.guess_format does support this now after changes you made to it:

https://github.com/RDFLib/rdflib/blob/9379a69d6ec6e18819aa0c5a0d54849e7abb8223/rdflib/util.py#L354-L370

Reply:
Yes, I think so, so I will remove this and test.